Feat icon

Amber-Song opened the pull request:
# Context

Link to [Notion ticket](https://www.notion.so/cloudcannon/Icon-a3273cf7d4fe4d6d867332e1c93b36ef?pvs=4)

# Additional information

The icon doesn't work well for bookshop. To see how the icon is, add the front matter to pages like:
In `index.html`
```
    - _bookshop_name: generic/icon
      svg_icon: solid/academic-cap
      big_icon: false
      rounded_border: false
```
Then you can see the icon in index page in your local machine. You still not be able to see it in cloud cannon site because it uses bookshop, but you can see it in live site.

* Never mind: Above is changed.

# Testing (tester)

The reviewer should check on this branch:

- The code
- The component in index page
- The site in CloudCannon ([link here](https://vital-caterpillar.cloudvent.net/))

# Before merge (PR owner)

- Delete the test site in CloudCannon

Commit:
Feat icon

Commit:
Updated 1 file via CloudCannon.

Commit:
Updated 1 file via CloudCannon.

Commit:
Update icon

Commit:
Merge branch 'feat/icon' of github.com:CloudCannon/small-business-template-eleventy into feat/icon

Commit:
Updated 2 files via CloudCannon.

Commit:
Updated 1 file via CloudCannon.

Commit:
Correct code

Commit:
Fix icon in bookshop

Commit:
Change px to rem

aleksandrgordeev commented on src/pages/index.html:
```diff
@@ -1,22 +1,25 @@
+        _bookshop_name: generic/icon
+        svg_icon: outline/arrow-long-up
+        big_icon: true
+        rounded_border: true
```
Maybe we should discard these changes? It looks like a test output.

James-cloudcannon replied:
Yeah I assume this was just in place for testing the component. It makes it easier for us to see but we can just add this manually ourselves if instead of pushing this if needed.

Amber-Song replied:
I will make sure to delete it before merge.

aleksandrgordeev commented on src/pages/booking.html:
```diff
@@ -7,4 +7,4 @@
 
-<h1>booking</h1>
\ No newline at end of file
+<h1>booking</h1>
```
What is the purpose of this change?

James-cloudcannon replied:
this was just code from the boiler plate, looks like a commit removed it and then added back? These lines on each page can be deleted completely.

aleksandrgordeev commented on component-library/components/generic/icon/icon.bookshop.yml:
```diff
@@ -0,0 +1,613 @@
+# Defines the structure of this component, as well as the default values
+blueprint:
+  svg_icon:
+  big_icon: true
```
This parameter is not in the ticket description. May we add this to the ticket description?

Amber-Song replied:
Yes. I will add it in the ticket.

aleksandrgordeev reviewed (changes requested):
The component works well 👍

I've added a couple of small suggestions. And there is a merge conflict that needs to be resolved.

James-cloudcannon commented on component-library/components/generic/icon/icon.scss:
```diff
@@ -0,0 +1,24 @@
+.c-icon {
+    &__round {
+        border-radius: 50%;
+        border-color: var(--neutral-000, #FFF);
```
--neutral-000 needs to be changed to --brand-grey-000

Amber-Song replied:
Thanks.

James-cloudcannon commented on component-library/components/generic/icon/icon.scss:
Can't seem to get the border-color to work. As a simple workaround you can remove `border-color: transparent `out of __big and __small and adding something like this to each of those classes:

`&:not(.c-icon__round) {
            border-color: transparent;
        }`

Amber-Song replied:
Oh, that works! Amazing! I almost try to hack it with !important. 😅

James-cloudcannon commented on component-library/components/generic/icon/icon.eleventy.liquid:
Seems to work well, assuming that the colour is inherited from the component using this.

Amber-Song replied:
I tried on --brand-grey-900 and it works properly.

James-cloudcannon commented on component-library/components/generic/icon/icon.scss:
Might need to work on the padding & width/height. Small should be 32x32 pixels. Large should be 69x69 pixels

Amber-Song replied:
Cool.

James-cloudcannon replied:
Sorry, should of made this one clearer! The above sizes should also include the padding. Currently the svg is set to 32x32px and 69x69px with the addition of 16px padding.

Amber-Song replied:
Are we using rem or px?

James-cloudcannon reviewed (commented):
Nice work! just a few small changes to do and then we should be able to get this merged.

Commit:
Add icon_fill_color in

Commit:
Fix the border style and color

Commit:
Recalculate the w and h

Commit:
Test on index

Commit:
Merge branch 'main' into feat/icon

Commit:
Recalculate h and w again

Commit:
Remove test component

Amber-Song commented:
> The component works well 👍
> 
> I've added a couple of small suggestions. And there is a merge conflict that needs to be resolved.

Yes. I have already changed those.

Amber-Song commented:
> just a few small changes to do and then we should be able to get this merged.

Done! Thanks for the review. I learnt a lot.

Commit:
Comment on color on image

Commit:
Remove content block

aleksandrgordeev reviewed (approved):
Nice work 👍

AlyshaNolan committed:
Test: CloudCannon configuration

AlyshaNolan committed:
Chore: remove CloudCannon configuration test

AlyshaNolan committed:
Fix: input names

AlyshaNolan committed:
Fix: BEM convention

AlyshaNolan committed:
Feat: allow the icon color to be changed in bookshop

AlyshaNolan committed:
Chore: test CloudCannon configuration of component

AlyshaNolan committed:
Fix: add back rounded border class

AlyshaNolan committed:
Feat: update blueprint and change border widths

AlyshaNolan committed:
Fix: border width and add back in fill/outline style

AlyshaNolan committed:
Fix: sizes and remove :not selector

AlyshaNolan committed:
Updated 1 file via CloudCannon.

AlyshaNolan committed:
Test: stylesheet not being included

AlyshaNolan committed:
Chore: remove test

AlyshaNolan committed:
Test: build sass

AlyshaNolan committed:
Updated 1 file via CloudCannon.

AlyshaNolan committed:
Chore: remove structure and improve description

AlyshaNolan commented:
Made a couple of changes for this one:
- The classes in the top level div were using __ when they should've been using --
- Change big_icon input name to be more editor friendly
- Change implementation for icon color when in bookshop so it can be seen
- Change color input defaults to use hex codes
- Fix styles not being generated on the live site
